packages/block-editor/src/components/block-list/block.js

```diff
@@ -544,7 +546,15 @@ function BlockListBlock( {
 						ref={ breadcrumb }
 					/>
 				) }
-				{ ( shouldShowContextualToolbar || isForcingContextualToolbar.current ) && (
+				{ ( ( ! consumeChildToolbar && shouldShowContextualToolbar ) || isForcingContextualToolbar.current ) && (
+					<BlockContextualToolbar
```

Comment:
This component and the comments are duplicating the one we have bellow. Could we join both conditions with an or and avoid this code duplication?

Reply:
Agreed. There's no point in doubling up. It's currently like it is for clarity during dev. But we need to consolidate and DRY it up.

Reply:
I think this has now become redundant due to refactoring.